tests/test_cloudbuild_config.py

```
@@ -0,0 +1,566 @@
+            self.assertEqual(st.get("name"), "gcr.io/google.com/cloudsdktool/cloud-sdk")
+            args = st.get("args", [])
+            self.assertIsInstance(args, list)
+            self.assertIn("functions", args)
+            self.assertIn("deploy", args)
+            self.assertIn("event-parser", args)
+            self.assertIn("--runtime=nodejs20", args)
+            self.assertIn("--entry-point=parseAgentEvent", args)
+            # Ensure service account param is present
+            self.assertTrue(any(a.startswith("--service-account=") and "event-parser-sa@" in a for a in args))
+
+for k in expected:
```

Ruff (0.12.2)

Undefined name `expected`

(F821)